src/components/Image/ImageLoader.jsx

```diff
@@ -1,46 +1,141 @@
-import React, { Component } from 'react'
+import React, { PureComponent } from 'react'
 import { getDownloadLink } from 'cozy-client'
 
-export default class ImageLoader extends Component {
+const TTL = 6000
+
+const LOADING = 'LOADING'
+const LOADED = 'LOADED'
+const PENDING = 'PENDING'
+const FAILED = 'FAILED'
+
+export default class ImageLoader extends PureComponent {
   state = {
-    fallback: null
+    fallbackUrl: null,
+    primaryStatus: LOADING,
+    fallbackStatus: PENDING
+  }
+
+  isLoadingPrimary() {
+    return this.state.primaryStatus === LOADING
+  }
+  wantsToLoadPrimary() {
+    return this.isLoadingPrimary() && !this.img
+  }
```

Comment:
Cette fonction retourne un bouléen, du coup, je me dis qu'elle pourrait commencer par `is` ou `has` ou `should`.

Reply:
:+1: pour le `should`